src/core/Core.js

```diff
@@ -394,7 +395,7 @@ class Uppy {
    */
   generatePreview (file) {
     if (Utils.isPreviewSupported(file.type) && !file.isRemote) {
-      Utils.createThumbnail(file, 200).then((thumbnail) => {
+      Utils.createThumbnail(file, 200, this.opts.thumbnailGeneration).then((thumbnail) => {
```

Comment:
I wonder if it would be better to do the `this.opts.thumbnailGeneration` check here instead of changing `createThumbnail`? You probably wanted to keep it clean with the promise, maybe that’s a good idea.

Reply:
Yeah as you say, it felt cleaner to do it in the createThumbnail method, otherwise the promise gets a bit complicated. As I wanted it to set the value of preview to be the current image it meant it had to go through the URL.createObjectURL method, which was already part of the createThumbnail method.

That said, happy to go either way.. your call!

Reply:
It's true that the promise would be a bit more complicated but I think it would make more sense for the `opts.thumbnailGeneration` check to happen here, with one branch generating a thumbnail and the other only doing `URL.createObjectURL`. the method names make sense that way: we're generating a preview, which can be a thumbnail, or just the image itself.

Reply:
No worries, I'll update the branch shortly 👍